crates/ide/src/syntax_highlighting/test_data/highlight_unsafe.html

```diff
@@ -121,6 +121,6 @@
         <span class="keyword">let</span> <span class="struct">Packed</span> <span class="brace">{</span> <span class="field">a</span><span class="colon">:</span> <span class="keyword unsafe">ref</span> <span class="variable declaration reference">_a</span> <span class="brace">}</span> <span class="operator">=</span> <span class="variable">packed</span><span class="semicolon">;</span>
 
         <span class="comment">// unsafe auto ref of packed field</span>
-        <span class="variable">packed</span><span class="operator">.</span><span class="field">a</span><span class="operator">.</span><span class="function associated reference trait unsafe">calls_autoref</span><span class="parenthesis">(</span><span class="parenthesis">)</span><span class="semicolon">;</span>
```

Comment:
This line https://github.com/rust-lang/rust-analyzer/blob/434e718b6717246066898a2dfcda1aec5e6f4082/crates/ide/src/syntax_highlighting/highlight.rs#L513
needs to use `it.containing_trait_or_trait_impl(sema.db)`